### PR TITLE
Add Diffblue tests

### PR DESCRIPTION
**Pull request by sudo-jaa**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -84,7 +84,7 @@
 			</plugin>
 
 			<plugin>
-				<!-- In the future, rather than using ModiTect to compile module-info.java, 
+				<!-- In the future, rather than using ModiTect to compile module-info.java,
```
Comment (JohnLBergqvist) [inline pom.xml]: Fix.

File: pom.xml [hunk 2]
```diff
@@ -346,6 +346,26 @@
...
+		  <scope>test</scope>
+		</dependency>
+
+		<dependency>
+		   <groupId>org.powermock</groupId>
+		   <artifactId>powermock-api-mockito-common</artifactId>
+		   <version>1.6.5</version>
+		</dependency>
+
+		<dependency>
```
Comment (JohnLBergqvist) [inline pom.xml]: Fix alignment so it matches the other dependencies

File: pom.xml [hunk 3]
```diff
@@ -413,7 +433,7 @@
 										<goal>testCompile</goal>
 									</goals>
 									<configuration>
-										<!-- fork is needed if using different java versions for main and 
+										<!-- fork is needed if using different java versions for main and
```
Comment (JohnLBergqvist) [inline pom.xml]: Fix

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.File;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({"javax.management.*"})
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/utils/JarUtilsTest.java]: All this powermock & mockito & RunWith stuff is unnecessary. Also, i'd probably remove the timeout stuff in every test too. And remove the powermock & mockito dependencies from `pom.xml`

File: src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+public class ReferenceEqualityKeyTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  @Rule public Timeout globalTimeout = new Timeout(10000);
+
+  /* testedClasses: ReferenceEqualityKey.java */
+
+  @Test
+  public void hashCodeOutputNegative() {
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java]: I don't think people will accept tests `hashCode`. it's literally testing the `.hashCode()` function of an object. Given that nothing we've seen has existing tests for it, it's probably not worth covering, from the point of view of an open-source dev.

File: src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    final ReferenceEqualityKey objectUnderTest = new ReferenceEqualityKey(null);
+    final ReferenceEqualityKey other = new ReferenceEqualityKey(null);
+
+    // Act
+    final boolean retval = objectUnderTest.equals(other);
+
+    // Assert result
+    Assert.assertEquals(true, retval);
+  }
+}
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java]: remember to add the newline at the end of the file

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
All tests are fine unless i've commented on them.

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java (hunk 4 above)
Comment (sudo-jaa) [reply src/test/java/io/github/classgraph/utils/JarUtilsTest.java]: What does it mean when we @RunWith powermock? I may be mistaken but this annotation doesn't occur in every test that uses powermock, correct?

File: pom.xml [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 	<groupId>io.github.classgraph</groupId>
 	<artifactId>classgraph</artifactId>
-	<version>4.4.1-SNAPSHOT</version>
+	<version>4.4.2-SNAPSHOT</version>
```
Comment (sudo-jaa) [inline pom.xml]: This is the up-to-date pom from the origin. This won't appear in the diff when we make the real PR.

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class JarUtilsTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  /* testedClasses: JarUtils.java */
+
+  @Test
```
Comment (paulkennethkent) [inline src/test/java/io/github/classgraph/utils/JarUtilsTest.java:35]: Do they have full coverage on the rest of this method? Is this test covering an edge case that they didn't think of?

**Review by petersear — APPROVED** [review]

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java (hunk 8 above)
Comment (sudo-jaa) [reply src/test/java/io/github/classgraph/utils/JarUtilsTest.java:35]: It is unfortunately the only test we got for this method. Couldn't detect existing coverage, but they don't appear to have any existing tests for this class.

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java (hunk 4 above)
Comment (JohnLBergqvist) [reply src/test/java/io/github/classgraph/utils/JarUtilsTest.java]: I thought it did. But if you're not using Powermock, then you don't need it, obviously.

File: src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+import io.github.classgraph.json.ReferenceEqualityKey;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class ReferenceEqualityKeyTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  /* testedClasses: ReferenceEqualityKey.java */
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/json/ReferenceEqualityKeyTest.java]: I've removed this comment as well on my tests.

File: src/test/java/io/github/classgraph/utils/JarUtilsTest.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+import io.github.classgraph.utils.JarUtils;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class JarUtilsTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  /* testedClasses: JarUtils.java */
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/utils/JarUtilsTest.java]: See above comment.

File: src/test/java/io/github/classgraph/utils/WhiteBlackListTest.java [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+import io.github.classgraph.utils.WhiteBlackList;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class WhiteBlackListTest {
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+  /* testedClasses: WhiteBlackList.java */
```
Comment (JohnLBergqvist) [inline src/test/java/io/github/classgraph/utils/WhiteBlackListTest.java]: See above comment.

**Review by JohnLBergqvist — CHANGES_REQUESTED** [review]
Minor fixes.
